src/frontend/src/pages/index.js

```diff
@@ -14,14 +41,23 @@ export default function IndexPage() {
   const { telescopeUrl } = useSiteMetaData();
   const savedCallback = useRef();
 
+  // Pagination
+  const [nextPageLink, setNextPageLink] = useState(`/posts?page=${numPages}`);
+  const [loading, setLoading] = useState(false);
+  const classes = useStyles();
+
   useEffect(() => {
-    async function getPosts(pageNum = 1) {
+    async function getPosts() {
       try {
-        const res = await fetch(`${telescopeUrl}/posts?page=${pageNum}`);
+        setLoading(true);
+        const res = await fetch(`${telescopeUrl}${nextPageLink}`);
+
         if (!res.ok) {
           throw new Error(res.statusText);
         }
+
         const postsUrls = await res.json();
+        const links = parse(res.headers.get('Link'));
         const postsData = await Promise.all(
           postsUrls.map(async ({ url }) => {
```

Comment:
Let's shorten this:

```js
postUrls.map(({url}) => fetch(`${telescopeUrl}${url}`).then(res => res.json()));
```

The overarching error handling can deal with any other issues in here.

Reply:
I'm not sure this replaces what we have, as we need the   const res = await fetch(`${telescopeUrl}${nextPageLink}`); to get the response to get      const postsUrls = await res.json(); &&  const links = parse(res.headers.get('Link'));

Reply:
I'm not suggesting touching the rest of the outer function, it was the part from 62-68 only.  You don't need to `await` in there, and can return the Promise

Reply:
Okay fixed.